.github/workflows/publish-charm.yaml

```diff
@@ -0,0 +1,41 @@
+name: publish-charm
+
+on:
+  workflow_call:
+
+jobs:
+  publish-charm:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install charmcraft
+        run: sudo snap install charmcraft --classic
+      - name: Fetch Tested Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: tested-charm
+      - name: Move charm in current directory
+        run: find ./ -name sdcore-smf_ubuntu-22.04-amd64.charm -exec mv -t ./ {} \;
+      - name: Select Charmhub channel
+        uses: canonical/charming-actions/channel@2.2.5
+        id: channel
+      - name: Upload charm to Charmhub
+        uses: canonical/charming-actions/upload-charm@2.2.5
+        with:
+          credentials: "${{ secrets.CHARMCRAFT_AUTH }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          upload-image: "true"
+          channel: "${{ steps.channel.outputs.name }}"
+
+      - name: Chmod charmcraft logs
+        if: steps.upload.conclusion == 'failure'
+        run: |
+          sudo chmod -R 777 /root/snap/charmcraft/common/cache/charmcraft/log/
+
+      - name: Archive charmcraft logs
+        if: steps.upload.conclusion == 'failure'
+        uses: actions/upload-artifact@v3
+        with:
+          name: charmcraft-upload-logs
+          path: /root/snap/charmcraft/common/cache/charmcraft/log/*.log
```

Comment:
Not required with the changes requested above

Reply:
Removed.